src/App.tsx

```diff
@@ -1,10 +1,61 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+
 import { USER_ID } from './api/todos';
+import { todos as todosClient } from './api/todos';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { TodoHead } from './Components/TodoHead';
+import { TodoList } from './Components/TodoList';
+import { TodoFilter } from './Components/TodoFilter';
+import { Errors } from './types/Errors';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.all);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
```

Comment:
Do you need separate state for filtered todos? If you can use only one state `todos`
```suggestion
```

Reply:
Used to store todos from server to avoid api call when filtering. Same as Example. Is there better way to do it?